e2e/next-sandbox/test/comments/index.test.ts

```diff
@@ -22,21 +22,25 @@ test.describe("Comments", () => {
   test.afterEach(() => Promise.all(pages.map((page) => page.close())));
 
   test("verify A and B display same number of threads after threads are loaded", async () => {
-    await waitForJson(pages, "#isLoading", false);
-    await waitUntilEqualOnAllPages(pages, "#numOfThreads");
+    await waitForJson(pages, "#isLoading", false, { timeout: 15_000 });
+    await waitUntilEqualOnAllPages(pages, "#numOfThreads", { interval: 250 });
   });
 
   test("verify thread creation on B is broadcasted correctly to A", async () => {
     const [page1, page2] = pages;
 
-    await waitForJson(pages, "#isLoading", false);
+    await waitForJson(pages, "#isLoading", false, { timeout: 15_000 });
+    await waitUntilEqualOnAllPages(pages, "#numOfThreads", { interval: 250 });
 
-    const numOfThreads = (await getJson(page1, "#numOfThreads")) as number;
+    // Read starting value n
+    const n = (await getJson(page1, "#numOfThreads")) as number;
 
+    await page1.click("#create-thread");
     await page2.click("#create-thread");
-    await waitForJson(page1, "#numOfThreads", numOfThreads + 1);
+    await page2.click("#create-thread");
+    await waitForJson(page1, "#numOfThreads", n + 3, { timeout: 15_000 });
 
     await page2.click("#delete-comment");
-    await waitUntilEqualOnAllPages(pages, "#numOfThreads");
+    await waitForJson(pages, "#numOfThreads", n + 3 - 1, { timeout: 15_000 });
```

Comment:
I could have written `n + 2` here, but hopefully this makes it a bit more readable.